cpp/include/raft/linalg/mdarray.h

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <experimental/mdspan>
+#include <raft/cudart_utils.h>
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <thrust/device_ptr.h>
+
+namespace raft::linalg {
```

Comment:
We're planning for these to go in what's ultimately going to be the `raft::core` package.

Reply:
Thanks for sharing, but I thought raft is the `core`? ;-)

Reply:
What I mean is that the mdarray  belongs the `raft/` directory / namespace and not in `linalg`

Reply:
Got it, let me change it.

Reply:
Move the file into `raft/` and removed the `linalg` namespace.